### DISCUSSION TITLE
Use code server behind nginx

### DISCUSSION THREAD

**Discussion by piatho** [coder/code-server#5125] [Q&A]
Dear community,
i have setup my codeserver instance as a container following the offical documentation. Accessing the container via http works fine for me. Everything works as expected. However as soon as i use my nginx reverse proxy manager (GUI) to forward mydomain.com via an encrypted connection to the code server (https to http), i cannot switch themes. Although the settings seems to be correcly saved with a black theme, i am still getting the white default theme.

I have the feeling this[Helper-Page](https://github.com/coder/code-server/blob/main/docs/guide.md#self-signed-certificate) already tells me what to do but it looks like I need a more simplistic explanation. 

Browser: Chrome

Appreciate your help :-)

**Comment by jsjoeio** [comment]
Hmm...does this happen with all themes?

I think the theme is saved locally in the browser 🤔 I'm not sure why it wouldn't work using a reverse proxy. Have you tried using another proxy like Caddy to see if you have the same issue?

**Reply by piatho** [reply]
Hi @jsjoeio, i do not have an instance of Caddy available in my home environment. However, i noticed that also extensions such as vscode-icons are not working via an encrypted connection and nginx. Can it be a false configuration of the container, Do i need to specify the domain somewhere in the container configuration?

**Reply by jsjoeio** [reply]
Argh! It could be...Unfortunately I have never used nginx so I'm not sure how much help I can be. I've seen a lot of nginx issues lately.

Possibly 🤔

Are there any good tutorials/docs you recommend on using nginx? I wonder if I could try using it locally on my macOS + Brave + code-server to see if I can help more.

**Reply by piatho** [reply]
Well, i have made my life here very simple. I have simply followed the instructions on https://nginxproxymanager.com/guide/#quick-setup to setup my container with nginx. Once spinned up, i have never left the graphical user interface. The interface is very simple. You specify a domain (scheme https) and a Forward Hostname / IP (my code-server docker instance) (scheme http) and request a lets encrypt certificate.

**Reply by piatho** [reply]
Reviewing the chrome console, i noticed that all resources that do some layouting are not accessably via the reverse proxy:
- https://mydomain/vscode-remote-resource?path=%2Fapp%2Fcode-server%2Flib%2Fvscode%2Fextensions%2Ftheme-defaults%2Fthemes%2Fdark_vs.json&tkn=
- https://mydomain.com/vscode-remote-resource?path=%2Fconfig%2Fextensions%2Fvscode-icons-team.vscode-icons-11.11.0%2Fdist%2Fsrc%2Fvsicons-icon-theme.json&tkn= (error 404)

**Reply by jsjoeio** [reply]
Interesting! Can you share logs or screenshots for those 404? Maybe anything helpful in the Network tab too?

I wonder why those aren't accessible.

**Reply by piatho** [reply]
I did spend some time understanding the concepts. Finally, i decided for a solution that isn't 100% what i preferred but which works. 

What does not work are custom locations. In NGINX you can define for one domain multiple custom locations like mydomain.com/vscode to be proxied to container 1, which mydomain.com/sql is proxied somewhere else (see screenshot attached)
![image](https://user-images.githubusercontent.com/68991540/165480223-dd085e67-6580-45c0-97c8-686ee115c3e5.png)

In this setup non of my containers work. All through errors on missing static files. It seems to require additional configuration on the host (here the code server) to set up a virtual host plus additional advanced redirection directives.

Finally, i ended up using a subdomain instead. so vscode.mydomain.com works fine.

**Reply by jsjoeio** [reply]
> it seems to require additional configuration on the host (here the code server)

Interesting!

> Finally, i ended up using a subdomain instead. so vscode.mydomain.com works fine

Well, I'm really happy you found a solution! Sorry I couldn't help more.
